dissect/evidence/ad1/ad1.py

```diff
@@ -0,0 +1,347 @@
+from __future__ import annotations
+
+import re
+import zlib
+from datetime import datetime, timezone
+from functools import cached_property
+from pathlib import Path, PurePosixPath, PureWindowsPath
+from typing import TYPE_CHECKING, BinaryIO
+
+from dissect.util.stream import AlignedStream, MappingStream, RelativeStream
+
+from dissect.evidence.ad1.c_ad1 import c_ad1
+from dissect.evidence.exceptions import FileNotFoundError, NotADirectoryError, NotASymlinkError
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+
+EntryType = c_ad1.EntryType
+MetaType = c_ad1.MetaType
+FileClassType = c_ad1.FileClassType
+
+
+def atoi(text: str) -> int | str:
+    return int(text) if text.isdigit() else text
+
+
+def natural_keys(text: str | Path) -> list[int | str]:
+    return [atoi(c) for c in re.split(r"(\d+)", str(text))]
+
+
+def find_files(path: Path) -> set[Path]:
+    files = set()
+    for file in path.parent.iterdir():
+        if file.stem == path.stem and re.match(r"^\.ad[0-9]+$", file.suffix.lower()):
+            files.add(file)
+    return sorted(files, key=natural_keys)
```

Comment:
```suggestion
    return sorted(files, key=lambda file: int(file.suffix[3:]))
```
Any reason why this wouldn't work?